dlt/normalize/items_normalizers.py

```diff
@@ -52,6 +63,120 @@ def __init__(
     def __call__(self, extracted_items_file: str, root_table_name: str) -> List[TSchemaUpdate]: ...
 
 
+class ModelItemsNormalizer(ItemsNormalizer):
+    def _add_or_replace_column(
+        self,
+        parsed_select: sqlglot.exp.Select,
+        column_name: Literal["_dlt_load_id", "_dlt_id"],
+        alias_expression: sqlglot.exp.Alias,
+        root_table_name: str,
+    ) -> Optional[TSchemaUpdate]:
+        """
+        Adds or replaces a column in the SELECT clause of a parsed SQL query and
+        returns a single TSchemaUpdate if a new column was added, otherwise None.
+        """
+        schema = self.schema
+        schema_update: TSchemaUpdate = {}
```

Comment:
I would raise here if a star select is detected. Add a new exception that explains why dlt_id and dlt_load_id can't be processed if not all column names are known.